server/svix-server/src/v1/endpoints/endpoint/mod.rs

```diff
@@ -160,6 +163,19 @@ pub struct EndpointIn {
     pub key: Option<EndpointSecret>,
 }
 
+impl EndpointIn {
+    pub fn take_or_generate_key(
+        &mut self,
+        cfg: &Configuration,
```

Comment:
I'd be explicit as for what it receives. Encryption and default signature type, rather than passing the whole config.